### PR TITLE
Feature/cache check

### PR DESCRIPTION
**Pull request by toumorokoshi**
Fixes the caching bug, which causes carbon-cache to not be queries when there is a symlinked directory.

### CODE REVIEW

File: webapp/tests/settings.py [hunk 1]
```diff
@@ -51,3 +51,5 @@ def atexit_tmpremover(dirname):
 INDEX_FILE = os.path.join(TEMP_GRAPHITE_DIR, 'index')
 
 URL_PREFIX = '/graphite'
+
+CARBON_CACHE_FINDER = 'tests.fixtures.EmptyFinder'
```
Comment (toumorokoshi) [inline webapp/tests/settings.py:55]: doing this fixes some unit tests that were failing with the carbon cache finder.

File: requirements.txt [hunk 2]
```diff
@@ -44,7 +44,7 @@ simplejson==2.1.6
 django-tagging==0.4.3
 gunicorn
 pytz
-pyparsing
+pyparsing==2.2.0
```
Comment (toumorokoshi) [inline requirements.txt:47]: we can remove this pin, by pyparsing 1.5.3 wasn't working.

**Comment by toumorokoshi** [comment]
Some context on the fix:

when whisper metrics were written to a symlink directory, the get_real_metric_path method would incorrect strip a top level directory, causing metrics like "foo.bar" to just become "bar".

This fixes that behavior.

**Comment by yunstanford** [comment]
LGTM

**Pull request merged**